test/lib/ufe/testContextOps.py

```diff
@@ -157,6 +158,14 @@ def testSwitchVariantInLayer(self):
         contextItemStrings = [c.item for c in contextItems]
         self.assertIn('Variant Sets', contextItemStrings)
 
+        # Add an attribute in the session layer to see if it affects
+        # switching variant.
+        stage = self.ball35Prim.GetStage()
+        stage.SetEditTarget(stage.GetSessionLayer())
+        self.ball35Prim.GetAttribute("xformOpOrder").Set(Vt.TokenArray("translate"))
+
+        stage.SetEditTarget(stage.GetRootLayer())
+
```

Comment:
This change to the existing test for switching in different layers reproduced the reported problem by authoring an unrelated attribute in the session layer. It prevented authoring when we were applying too strict restrictions. Now we are only checking that the desired metadata has no no stronger opinion.